node/config/load.go

```diff
@@ -17,17 +18,40 @@ import (
 
 // FromFile loads config from a specified file overriding defaults specified in
 // the def parameter. If file does not exist or is empty defaults are assumed.
-func FromFile(path string, def interface{}) (interface{}, error) {
+func FromFile(path string, def interface{}, opts ...LoadCfgOpt) (interface{}, error) {
+	var loadOpts cfgLoadOpts
+	for _, opt := range opts {
+		if err := opt(&loadOpts); err != nil {
+			return nil, xerrors.Errorf("failed to apply load cfg option: %w", err)
+		}
+	}
+
+	// check for loadability
 	file, err := os.Open(path)
+	defer file.Close() //nolint:errcheck // The file is RO
 	switch {
 	case os.IsNotExist(err):
+		if loadOpts.canFallbackOnDefault != nil {
+			if err := loadOpts.canFallbackOnDefault(); err != nil {
+				return nil, err
+			}
+		}
 		return def, nil
```

Comment:
Ah, sorry, I wasn't clear. We have two options to control "defaults", we should have a single option. E.g., something like:

```
func DefaultConfig(func() (interface{}, error))
```

This code here would then read:

```go
if loadOps.defaultConfig != nil {
    return loadOps.defaultConfig()
}
return nil, xerrors.Errorf("no config found")
```

Reply:
That makes sense and I tried to do it.  The problem is we have two different uses of the default.  One to fallback on if the config file is not found and one for us to use to set the defaults when reading the existing config from disk.  This change wants two different configuration options: a default config to pass in to the function populating the config and a way to error if there is no config specified on disk

Reply:
I'll add the default value as a separate functional option though